pkg/iscsi/internal_kubernetes.go

```
@@ -0,0 +1,800 @@
+	kubeletRegistrationPath = strings.TrimSpace(kubeletRegistrationPath)
+	if kubeletRegistrationPath == "" {
+		return fmt.Errorf("kubelet registration path is required")
+	}
+	socketPath := filepath.Join("/registration", d.name+"-reg.sock")
+	if err := os.Remove(socketPath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("remove stale registration socket %q: %w", socketPath, err)
+	}
+	if err := os.MkdirAll(filepath.Dir(socketPath), 0o755); err != nil {
+		return fmt.Errorf("create registration socket directory: %w", err)
+	}
+	listener, err := net.Listen("unix", socketPath)
```

golangci-lint (2.12.1)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)